client/state/current-user/gravatar-status/reducer.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	GRAVATAR_UPLOAD_RECEIVE,
+	GRAVATAR_UPLOAD_REQUEST,
+	GRAVATAR_UPLOAD_REQUEST_SUCCESS,
+	GRAVATAR_UPLOAD_REQUEST_FAILURE
+} from 'state/action-types';
+import { createReducer } from 'state/utils';
+
+export const isUploading = createReducer( false, {
+	[ GRAVATAR_UPLOAD_REQUEST ]: () => true,
+	[ GRAVATAR_UPLOAD_REQUEST_SUCCESS ]: () => false,
+	[ GRAVATAR_UPLOAD_REQUEST_FAILURE ]: () => false
+} );
+
+export const tempImage = createReducer( {}, {
+	[ GRAVATAR_UPLOAD_RECEIVE ]: ( state, action ) => {
+		return {
+			src: action.src
+		};
+	}
+} );
```

Comment:
more importantly, why is the state here an object? are we adding more properties than `src`? if not, using a string would have significant benefits. each time we return this new object we are creating an entirely new object, even if the `src` is the same.

**with a string**
```js
// immutable and shallow-render friendly
export const tempImageSrc = ( state = '', { type, src } ) =>
	GRAVATAR_UPLOAD_RECEIVE === type
		? src
		: state;
```

**working around the object stuff**
```js
export const tempImage = ( state = {}, { type, src } ) =>
	GRAVATAR_UPLOAD_RECEIVE === type && src !== state.src
		? { src }
		: state;
```